pkg/frontend/sku_validation.go

```diff
@@ -11,15 +11,22 @@ import (
 	mgmtcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-01/compute"
 
 	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/compute"
 	"github.com/Azure/ARO-RP/pkg/util/computeskus"
 )
 
+type SkuValidator interface {
+	ValidateVMSku(ctx context.Context, location string, oc *api.OpenShiftCluster, resourceSkusClient compute.ResourceSkusClient) error
+}
+
+type skuValidator struct{}
+
 // ValidateWorkerSku uses resourceSkusClient to ensure that the VM sizes listed in the cluster document are available for use in the target region.
-func (dv *dynamic) ValidateVMSku(ctx context.Context, location string, subscriptionID string, oc *api.OpenShiftCluster) error {
+func (s skuValidator) ValidateVMSku(ctx context.Context, location string, oc *api.OpenShiftCluster, resourceSkusClient compute.ResourceSkusClient) error {
```

Comment:
we actually don't need the location string here, since it comes from the cluster object

Reply:
Removed location since we already use cluster